pkg/apis/extensions/v1beta1/defaults_test.go

```diff
@@ -280,6 +289,26 @@ func TestSetDefaultDeployment(t *testing.T) {
 	}
 }
 
+func TestDefaultDeploymentAvailability(t *testing.T) {
+	d := &Deployment{}
+	obj := roundTrip(t, runtime.Object(d))
+	in, ok := obj.(*Deployment)
+	if !ok {
+		t.Errorf("unexpected object: %v", in)
+		t.FailNow()
+	}
+
+	out := &extensions.Deployment{}
+	if err := api.Scheme.Convert(in, out, nil); err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+
+	maxUnavailable := deployutil.MaxUnavailable(*out)
+	if out.Spec.Replicas-maxUnavailable <= 0 {
+		t.Fatalf("rolling Deployment supports maximum unavailability by default")
```

Comment:
I think the message should be something like, `the default value of maxUnavailable leads to no active replicas during rolling update` or something like that. I dont understand the current message. Also in this test, what is the value of Spec.Replicas ?